peerdid/core/did_doc_types.py

```diff
@@ -0,0 +1,219 @@
+from enum import Enum
+from typing import List, Optional, Dict, Union, Tuple
+
+from peerdid.core.jwk_okp import get_verification_method_type
+from peerdid.errors import MalformedPeerDIDDocError
+from peerdid.types import (
+    VerificationMethodType,
+    VerificationMethodTypeAuthentication,
+    VerificationMethodTypeAgreement,
+    VerificationMaterial,
+    VerificationMaterialFormatPeerDID,
+    VerificationMaterialAuthentication,
+    VerificationMaterialAgreement,
+)
+
+
+class VerificationMethodField(Enum):
+    BASE58 = "publicKeyBase58"
+    MULTIBASE = "publicKeyMultibase"
+    JWK = "publicKeyJwk"
+
+
+SERVICE_ID = "id"
+SERVICE_TYPE = "type"
+SERVICE_ENDPOINT = "serviceEndpoint"
+SERVICE_DIDCOMM_MESSAGING = "DIDCommMessaging"
+SERVICE_ROUTING_KEYS = "routingKeys"
+SERVICE_ACCEPT = "accept"
+
+
+class VerificationMethodPeerDID:
+    def __init__(self, id: str, controller: str, ver_material: VerificationMaterial):
+        self.id = id
+        self.controller = controller
+        self.ver_material = ver_material
+
+    def to_dict(self):
+        return {
+            "id": self.id,
+            "type": self.ver_material.type.value,
+            "controller": self.controller,
+            self.public_key_field.value: self.ver_material.value,
+        }
+
+    @property
+    def public_key_field(self):
+        if self.ver_material.format == VerificationMaterialFormatPeerDID.BASE58:
+            return VerificationMethodField.BASE58
+        elif self.ver_material.format == VerificationMaterialFormatPeerDID.MULTIBASE:
+            return VerificationMethodField.MULTIBASE
+        elif self.ver_material.format == VerificationMaterialFormatPeerDID.JWK:
+            return VerificationMethodField.JWK
+        else:
+            raise ValueError(
+                "Unsupported verification material format "
+                + str(self.ver_material.format)
+            )
+
+    @classmethod
+    def _get_public_key_format(
+        cls, ver_method_type: VerificationMethodType
```

Comment:
here `@classmethod` might unnecessary and static one may be used, but not critical